source/time.tex

```diff
@@ -8039,713 +8077,340 @@
 
 \indexlibrary{\idxcode{operator/}!calendar types|)}
 
-\rSec1[time.tod]{Class template \tcode{time_of_day}}
+\rSec1[time.hms]{Class template \tcode{hh_mm_ss}}
 
-\rSec2[time.tod.overview]{Overview}
-\indexlibrary{\idxcode{time_of_day}}
+\rSec2[time.hms.overview]{Overview}
+\indexlibrary{\idxcode{hms}}
 
 \begin{codeblock}
 namespace std::chrono {
-  template<class Duration> class time_of_day;
-
-  template<> class time_of_day<hours>;
-  template<> class time_of_day<minutes>;
-  template<> class time_of_day<seconds>;
-  template<class Rep, class Period> class time_of_day<duration<Rep, Period>>;
-}
-\end{codeblock}
-
-\pnum
-The \tcode{time_of_day} class template
-splits a \tcode{duration} representing the time elapsed since midnight
-into a ``broken down'' time of day such as
-$hours$:$minutes$:$seconds$.
-The \tcode{Duration} template parameter dictates
-the precision to which the time is broken down.
-\begin{note}
-This can vary from a coarse precision of hours
-to a very fine precision of nanoseconds.
-\end{note}
-A \tcode{time_of_day} object also tracks
-whether it should be output
-as a 12-hour time format or a 24-hour time format.
+  template <class Duration>
+  class hh_mm_ss
+  {
+    bool            is_neg;     // \expos
+    chrono::hours   h;          // \expos
+    chrono::minutes m;          // \expos
+    chrono::seconds s;          // \expos
+    precision       ss;         // \expos
```

Comment:
This exposition-only member uses the typedef `precision` before it has been defined. Can the exposition-only members be moved to the end of the class?

Reply:
Or you could copy `public` and move to the top:
```
public:
    static unsigned constexpr fractional_width = _see below_;
    using precision                                              = _see below_;
```

Reply:
Fwiw, the reason I have come to appreciate the "data members at the top" style is that it informs the reader about the specification/behavior of the immediately following but invisible compiler-supplied special members:
```
~hh_mm_ss() = default;
hh_mm_ss(hh_mm_ss const&) = default;
hh_mm_ss& operator=(hh_mm_ss const&) = default;
hh_mm_ss(hh_mm_ss&&) = default;
hh_mm_ss& operator=(hh_mm_ss&&) = default;
```

Reply:
"Immediately following"? I don't follow. These implicit special members are produced at the closing brace of the class definition, not before.

Reply:
I've moved the exposition-only members to the bottom.